src/main/kotlin/info/novatec/docu/parser/element/CallActivityParser.kt

```diff
@@ -0,0 +1,51 @@
+package info.novatec.docu.parser.element
+
+import info.novatec.cbdg.models.elements.CallActivity
+import info.novatec.constants.BPMN_ATTRIBUTE_CALLED_ELEMENT
+import info.novatec.constants.BPMN_TAG_CALL_ACTIVITY
+import org.camunda.bpm.model.bpmn.instance.Process
+
+object CallActivityParser {
+
+    /**
+     * This method first puts the information of the CallActivities in two different lists.
+     *
+     * For the list callActivityFlowElementList, the whole process is filtered by the CallActivity tag. The list is filled by objects from type
+     * [org.camunda.bpm.model.bpmn.instance.FlowElement], which displays all CallActivities and provides the information about Id, Name and the Called Element.
+     * For the list callActivityDocumentationList, the callActivityFlowElementList is iterated to extract the documentation of each CallActivity and to put it
+     * in the callActivityDocumentationList, as the documentation itself is provided within a list and cannot be accessed directly as an attribute from the
+     * [org.camunda.bpm.model.bpmn.instance.FlowElement] object.
+     *
+     * Then, the elements of both lists callActivityFlowElementList and callActivityDocumentationList are merged to one list callActivityFinalList, which
+     * provides the CallActivities as [CallActivity] object with the information about Id, Name and the Called Element and Element Documentation, comprised in
+     * one object.
+     * Also, the version tag of the process is included in this object
+     *
+     * @param process The process from which all CallActivities are extracted to a [CallActivity] object and put in the returned list
+     *
+     * @return the list of CallActivities as [CallActivity] object with the information about Id, Name and the Called Element and Element Documentation,
+     * comprised in one object.
+     */
+    fun parseToCallActivityList(process: Process): MutableList<CallActivity> {
+        val callActivityFlowElementList = process.flowElements.filter { it.elementType.typeName.equals(BPMN_TAG_CALL_ACTIVITY) }
+        val callActivityDocumentationList: MutableList<String> = mutableListOf()
+        callActivityFlowElementList.forEach { callActivityDocumentationList.add(it.documentations.stream().findFirst().orElse(null)?.textContent.toString()) }
```

Comment:
use map instead of forEach to directly generate a non-mutable list

```
callActivityFlowElementList
   .map {
        CallActivity(
                        it.id, // Id of CallActivity
                        it.name, // Name of CallActivity
                        process.camundaVersionTag, // Version Tag of Process
                        it.documentations.findFirst().map{doc -> doc.textContent.toString()}.orElse(null), // Element Documentation of CallActivity
                        it.first.domElement.getAttribute(BPMN_ATTRIBUTE_CALLED_ELEMENT) // Called Element of CallAcitivity
                    )}